### ISSUE TITLE
Consultation on Windows training file modification

### ISSUE THREAD

**Issue by huidabao** [mit-han-lab/bevfusion#624]
Hello, since I don't have Ubuntu environment, I can train on Windows, but I can't use torchpack. Then make the following changes
:I found a lot of errors. Could you please help point out where the errors are? Thank you very much.
import argparse
import copy
import os
import random
import time
import yaml
import numpy as np
import torch
from mmcv import Config
from torchpack.environ import set_run_dir

from mmdet3d.apis import train_model
from mmdet3d.datasets import build_dataset
from mmdet3d.models import build_model
from mmdet3d.utils import get_root_logger, convert_sync_batchnorm, recursive_eval

def main():
    parser = argparse.ArgumentParser()
    parser.add_argument("config", metavar="FILE", help="config file")
    parser.add_argument("--run-dir", metavar="DIR", help="run directory")
    args, opts = parser.parse_known_args()

    # 读取配置文件并指定UTF-8编码
    with open(args.config, 'r', encoding='utf-8') as f:
        config_text = f.read()

    cfg_dict = yaml.safe_load(config_text)
    cfg = Config(cfg_dict)

    torch.backends.cudnn.benchmark = cfg.cudnn_benchmark
    torch.cuda.set_device(0)  # Assuming you are using the first GPU

    if args.run_dir is None:
        # 禁用自动设置运行目录功能
        set_run_dir('')
        cfg.run_dir = ''
    else:
        set_run_dir(args.run_dir)
        cfg.run_dir = args.run_dir

    # dump config
    if cfg.run_dir:
        cfg.dump(os.path.join(cfg.run_dir, "configs.yaml"))

    # init the logger before other steps
    timestamp = time.strftime("%Y%m%d_%H%M%S", time.localtime())
    if cfg.run_dir:
        log_file = os.path.join(cfg.run_dir, f"{timestamp}.log")
    else:
        log_file = f"{timestamp}.log"
    logger = get_root_logger(log_file=log_file)

    # log some basic info
    logger.info(f"Config:\n{config_text}")

    # set random seeds
    if cfg.seed is not None:
        logger.info(
            f"Set random seed to {cfg.seed}, "
            f"deterministic mode: {cfg.deterministic}"
        )
        random.seed(cfg.seed)
        np.random.seed(cfg.seed)
        torch.manual_seed(cfg.seed)
        if cfg.deterministic:
            torch.backends.cudnn.deterministic = True
            torch.backends.cudnn.benchmark = False

    datasets = [build_dataset(cfg.data.train)]

    model = build_model(cfg.model)
    model.init_weights()
    if cfg.get("sync_bn", None):
        if not isinstance(cfg["sync_bn"], dict):
            cfg["sync_bn"] = dict(exclude=[])
        model = convert_sync_batchnorm(model, exclude=cfg["sync_bn"]["exclude"])

    logger.info(f"Model:\n{model}")

    train_model(
        model,
        datasets,
        cfg,
        distributed=False,
        validate=True,
        timestamp=timestamp,
    )

if __name__ == "__main__":
    main()

**Comment by zhijian-liu** [comment]
Thank you for your interest in our project. This repository is no longer actively maintained, so we will be closing this issue. Please refer to the amazing implementation at [MMDetection3D](https://github.com/open-mmlab/mmdetection3d/tree/main/projects/BEVFusion). Thank you again!
